USING_WITH_PYRIGHT.md

```diff
@@ -0,0 +1,73 @@
+# Pyright and Pylance Together
+
+[Pyright](https://github.com/Microsoft/pyright) is an open source version of a Python type checker that forms the basis for Pylance. Both Pyright and Pylance can be used inside of an IDE to provide a [language server](https://code.visualstudio.com/api/language-extensions/language-server-extension-guide#why-language-server) for Python code.
+
+Pyright can also be used on the [command line](https://github.com/microsoft/pyright/blob/main/docs/command-line.md) to produce json or text output. This can be used in a variety of situations, like a [pre-commit hook](https://github.com/microsoft/pyright/blob/main/docs/ci-integration.md#running-pyright-as-a-pre-commit-hook), or a [github action](https://github.com/jakebailey/pyright-action).
+
+
+However, Pylance and Pyright don't always produce the same results. There are several reasons why this may happen.
+
+## Type stub differences
+
+Pylance comes bundled with a number of stubs, usually found here in the installation:
+
+```
+<pylance-extension-install>/dist/bundled
+<pylance-extension-install>/dist/native-stubs
+```
+
+For Pyright to produce the same errors as Pylance, it needs to use the same stub files. 
+
+This might be done by copying the contents of each of the bundled Pylance stub folders to a new folder. 
+
+You'd then reference that new folder in your [pyrightconfig.json](https://microsoft.github.io/pyright/#/configuration)
+
+```json
+{
+    "stubPath": "./pylanceStubs"
+}
+```
+
+or pyproject.toml
+
+```ini
+[tool.pyright]
+stubPath="./pylanceStubs"
+```
+
+Where `pylanceStubs` contains the contents of the `bundled` and `native-stubs` folders.
+
+We have plans to make this easier in the [future](https://github.com/microsoft/pylance-release/discussions/3638).
+
+## Settings differences
+
+Both Pylance and Pyright have [settings](https://github.com/microsoft/pylance-release#settings-and-customization) to control behavior. For the most part, they use the same values, but the few differences, described below, do have an impact on the analysis.
+
+
+| Setting | Pylance default | Pyright default | Description | Potential Impact |
+|----|----|----|----|----|
+| python.analysis.autoSearchPaths | true | false|  Adds 'src' to the list of search paths. | This may change what files are found when analyzing. So if you're getting missing imports for modules in your 'src' tree, this might be why. |
+| python.analysis.indexing | true | false | All files in the current virtual environment are parsed to a certain depth. This gives Pylance more information for features such as auto import. | This can have an impact on completions and hover information, but shouldn't impact errors returned.|

```

Comment:
```suggestion
| python.analysis.indexing | true | false | This setting enables all files in the current virtual environment to be parsed to a certain depth. This gives Pylance more information for features such as auto import. | This can have an impact on completions and hover information, but shouldn't impact errors returned.|

```